x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/TransactionLineChart/index.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo } from 'react';
+import { flatten } from 'lodash';
+import d3 from 'd3';
+import {
+  Coordinate,
+  RectCoordinate
+} from '../../../../../../typings/timeseries';
+import { useChartsTime } from '../../../../../hooks/useChartsTime';
+import { getEmptySeries } from '../../CustomPlot/getEmptySeries';
+// @ts-ignore
+import CustomPlot from '../../CustomPlot';
+import { toQuery, fromQuery } from '../../../Links/url_helpers';
+import { history } from '../../../../../utils/history';
+
+interface Props {
+  series: Array<{
+    color: string;
+    title: React.ReactNode;
+    titleShort?: React.ReactNode;
+    data: Array<Coordinate | RectCoordinate>;
+    type: string;
+  }>;
+  stacked?: boolean;
+  truncateLegends?: boolean;
+  tickFormatY: (y: number | null) => React.ReactNode;
+  formatTooltipValue: (c: Coordinate) => React.ReactNode;
+  yMax?: string | number;
+  height?: number;
+}
+
+const TransactionLineChart: React.FC<Props> = (props: Props) => {
+  const {
+    series,
+    tickFormatY,
+    formatTooltipValue,
+    stacked,
+    yMax,
+    height
+  } = props;
+
+  const flattenedCoordinates = flatten(series.map(serie => serie.data));
+
+  const start = d3.min(flattenedCoordinates, d => d.x);
+  const end = d3.max(flattenedCoordinates, d => d.x);
+
+  const noHits = series.every(
+    serie =>
+      serie.data.filter(value => 'y' in value && value.y !== null).length === 0
+  );
```

Comment:
In the rest of the code base `noHits` refers to whether Elasticsearch actually returned any hits. We cannot just look at `series.data` because we might want to serve some empty state data to the graph (to draw the x and y-axis). 

Therefore other chart endpoints return the `series.totalHits` (coming from ES `hits.total`):
https://github.com/elastic/kibana/blob/7056f6143d7863c2b48ff22ee376402c932bbb60/x-pack/legacy/plugins/apm/server/lib/metrics/transform_metrics_chart.ts#L58

I think we should do the same here if that makes sense to you?

Reply:
It would be more consistent, but I have some doubts about using hits.total to determine whether there is no data. Gut tells me there's a lot of cases where that will be a false positive. Using the actual data passed to the charts to determine if there are any data points by filtering out all non-null values feels like a better heuristic to me. Does that make sense?

Reply:
> Gut tells me there's a lot of cases where that will be a false positive. Using the actual data passed to the charts to determine if there are any data points by filtering out all non-null values feels like a better heuristic to me. Does that make sense?

That does make sense. My originally assumption was that if there are any hits then the aggregation will contain (timeseries) data, and it's been true for the aggs we do for the transaction charts. But this doesn't hold true for things like derivative aggs which requires at least two data points. So using `hits.total` as an indicator for data might not be a good idea.

If we go the direction you suggest (with a custom check) we should get rid of the `noHits`, right? Eg:
 - https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/apm/public/selectors/chartSelectors.ts#L55
 - https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/apm/public/components/app/ServiceDetails/MetricsChart.tsx#L37

I think it might make sense to go over how we handle this across the different charts. It seems like a pretty standard thing for a chart to be able to know whether it has data or not, and would be great if we can stick to one approach.